backend/program/indexers/trakt.py

```diff
@@ -77,17 +77,27 @@ def _add_seasons_to_show(show: Show, imdb_id: str):
 
 
 def _map_item_from_data(data, item_type: str, show_genres: List[str] = None) -> Optional[MediaItem]:
-    """Map trakt.tv API data to MediaItemContainer."""
+    """Map trakt.tv API data to MediaItemContainer.
+    
+    If the year is found in the slug, append the year to the title 
+    Slug is used as an page index for trakt but it's also a very good index for torrents
+    Seperating movies, shows of the same name.
+    """
     if item_type not in ["movie", "show", "season", "episode"]:
         logger.debug(f"Unknown item type {item_type} for {data.title} not found in list of acceptable items")
         return None
 
     formatted_aired_at = _get_formatted_date(data, item_type)
     genres = getattr(data, "genres", None) or show_genres
 
+    title = getattr(data, "title", None)
+    year = getattr(data, "year", None)
+    if hasattr(data.ids, "slug") and str(year) in getattr(data.ids, "slug"):

```

Comment:
I agree that this function needs improvement, but this isn't the way to do it

Reply:
Hey how you think I should go about this  ? 

Requesting the tmdb id data from track just tells us the title really however any search on the internet results in dark matter (2024) as such most torrents that relate to the 2024 version specify it in the torrent, should we make a well "known name" variable 




Data Dump -> 


[{"type":"show","score":1000,"show":{"title":"Dark Matter","year":2024,"ids":{"trakt":170597,"slug":"dark-matter-2024","tvdb":393159,"imdb":"tt19231492","tmdb":196322,"tvrage":null},"tagline":"One life. Infinite possibilities.","overview":"Jason Dessen is abducted into an alternate version of his life. To get back to his true family, he embarks on a harrowing journey to save them from the most terrifying foe imaginable: himself.","first_aired":"2024-05-08T02:00:00.000Z","airs":{"day":"Tuesday","time":"22:00","timezone":"America/New_York"},"runtime":42,"certification":"TV-MA","network":"Apple TV+","country":"us","trailer":"https://youtube.com/watch?v=j6ucGt_Xp14","homepage":"http://tv.apple.com/show/umc.cmc.4luj45vtqpmjsvb6sc2675oeg","status":"returning series","rating":7.206266318537859,"votes":383,"comment_count":24,"updated_at":"2024-06-13T15:22:36.000Z","language":"en","languages":["en"],"available_translations":["ar","bg","cs","da","de","el","en","es","fa","fi","fr","he","hi","hu","id","it","ja","ko","lt","nl","no","pl","pt","ru","sk","sl","sr","sv","th","tr","uk","vi","zh"],"genres":["fantasy","science-fiction","drama","mystery"],"aired_episodes":7}}]

Reply:
this function isnt for torrents. It maps data to an item based on an imdb

Reply:
My understanding is we receive an imdb_id from a requester(Overseerr, plex watchlist) and we do a trakt request to actually turn that id into a data item -> then we use that data item to scrap for torrents